pr_agent/algo/pr_processing.py

```diff
@@ -30,15 +32,17 @@ def get_pr_diff(git_provider: Union[GithubProvider, Any], token_handler: TokenHa
     pr_languages = sort_files_by_main_languages(git_provider.get_languages(), git_provider.pr.diff_files)
 
     # generate a standard diff string, with patch extension
-    patches_extended, total_tokens = pr_generate_extended_diff(pr_languages, token_handler)
+    patches_extended, total_tokens = pr_generate_extended_diff(pr_languages, token_handler,
+                                                               add_line_numbers_to_hunks)
```

Comment:
**Suggestion:** Consider using a more Pythonic way to handle optional parameters. Instead of using a boolean flag and then checking its value to decide whether to perform an action, you can use None as a default value and check if the parameter is None.
```suggestion
    def get_pr_diff(git_provider: Union[GithubProvider, Any], token_handler: TokenHandler,
                    add_line_numbers_to_hunks: Optional[bool] = None) -> str:
```